Sources/Validation/Validators/PhoneValidator.swift

```diff
@@ -0,0 +1,101 @@
+
+/// Describes the validation of the area code and the digits after the area code.
+/// Supports the following types:
+/// - **plain**: Only uses numbers. Example: `2397777777`
+/// - **dashWithParenthesis**: Uses Dash and Parenthesis. Example: `(239)777-7777`
+/// - **dashOnly**: Only uses Dashes. Example: `239-777-7777`
+public enum PhoneFormat {
```

Comment:
This should really be named `US/CanadaPhoneFormat`. I think this API could be better expressed with some notion of a country code. It would be nice to be able to do something like: 

```swift
try validations.add(\.name, .phone(.usa, .canada))
```

Reply:
I like that, is simple but takes away flexibility.
But my intentions with `PhoneFormat` was to give more flexibility to the format of the phone, I've seen that sometimes they use the parenthesis (777)653-0987 and others prefer to use the dash only 777-653-0987.

Maybe `PhoneType` could be the one that has the countries (If the name seems a little off I can work on changing it)

And then it will turn out to be:

```swift
try validations.add(\.phone, .phone(type: .usa(.dashWithParenthesis)))
```